### ISSUE TITLE
`Bug` HA crashes when alarm entity configured

### ISSUE THREAD

**Issue by russilker** [Blackymas/NSPanel_HA_Blueprint#2015]
### TFT Version

4.3.2

### ESPHome Version

4.3.2

### Blueprint Version

4.3.2

### Panel Model

US

### What is the bug?

As discussed in the [Alarmo github](https://github.com/nielsfaber/alarmo/issues/920), after upgrading to Home Assistant 2024.4, the entire Home Assistant instance crashes when the alarm state is changed. This only occurs if you have an alarm entity configured in the NSPanel blueprint; disabling the blueprint or removing the alarm entity from the blueprint stops the crash from happening.

### Steps to Reproduce

1. Set up the Alarmo integration
2. Add the Alarmo entity to the NSPanel blueprint as a custom button
3. Try arming the alarm. Home Assistant will instantly crash.

### Your Panel's YAML

```yaml
substitutions:

###### CHANGE ME START ######

  device_name: "nspanel-entryway" 
  wifi_ssid: !secret wifi_ssid
  wifi_password: !secret wifi_password

  nextion_update_url: "http://github.com/Blackymas/NSPanel_HA_Blueprint/raw/main/hmi/nspanel_us.tft" # URL to Github
  ##### CHANGE ME END #####



##### DO NOT CHANGE ANYTHING! #####

packages:
  ##### download esphome code from Github
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files: [nspanel_esphome.yaml]
    refresh: 300s

##### DO NOT CHANGE ANYTHING! #####

# Change framework to `esp-idf`
esp32:
  framework:
    type: esp-idf

# Enable Bluetooth proxy
bluetooth_proxy:
  active: TRUE
```

### Home Assistant Logs
```log
2024-04-05 12:23:32.314 INFO (MainThread) [custom_components.alarmo.alarm_control_panel] Alarm is now arming. Waiting for 60 seconds.
2024-04-05 12:23:32.435 ERROR (SyncWorker_54) [homeassistant.components.automation.nspanel_configuration] While executing automation automation.nspanel_configuration
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 726, in async_trigger
    return await self.action_script.async_run(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1650, in async_run
    return await asyncio.shield(create_eager_task(run.async_run()))
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 35, in create_eager_task
    loop=loop or get_running_loop(),
                 ^^^^^^^^^^^^^^^^^^
RuntimeError: no running event loop
2024-04-05 12:23:32.471 ERROR (MainThread) [homeassistant] Error doing job: Exception in callback Task.task_wakeup(<Future finished result=None>)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
     
NRuntimeError: Leaving task <Task pending name='Task-71' coro=<WebSocketHandler._writer() running at /usr/src/homeassistant/homeassistant/components/websocket_api/http.py:138> wait_for=<Future pending cb=[Task.task_wakeup()]>> does not match the current task <Task finished name="event trigger {'domain': 'automation', 'name': 'Entryway NSPanel Configuration', 'home_assistant_start': True, 'variables': {'this': {'entity_id': 'automation.nspanel_configuration', 'state': 'on', 'attributes': {'id': '1692629128632', 'last_triggered': datetime.datetime(2024, 4, 5, 15, 18, 2, 225595, tzinfo=datetime.timezone.utc), 'mode': 'parallel', 'current': 0, 'max': 50, 'friendly_name': 'Entryway NSPanel Configuration'}, 'last_changed': '2024-04-05T15:22:37.282131+00:00', 'last_reported': '2024-04-05T15:22:37.282131+00:00', 'last_updated': '2024-04-05T15:22:37.282131+00:00', 'context': {'id': '01HTQDNHV2WWZ4RTBVNE9Y4AR8', 'parent_id': None, 'user_id': None}}, 'nspanel_deviceid': '03c535246e9bbaa179317600cab1eb1e', 'nspanel_entities': ['binary_sensor.nspanel_entryway_left_button', 'binary_sensor.nspanel_entryway_right_button', 'binary_sensor.nspanel_entryway_nextion_display', 'sensor.nspanel_entryway_blueprint', 'sensor.nspanel_entryway_temperature', 'sensor.nspanel_entryway_display_current_brightness', 'sensor.nspanel_entryway_display_mode', 'sensor.nspanel_entryway_display_charset', 'sensor.nspanel_entryway_rssi', 'switch.nspanel_entryway_notification_unread', 'switch.nspanel_entryway_notification_sound', 'switch.nspanel_entryway_relay_1', 'switch.nspanel_entryway_relay_2', 'button.nspanel_entryway_restart', 'button.nspanel_entryway_update_tft_display', 'sensor.nspanel_entryway_detailed_entity', 'sensor.nspanel_entryway_current_page', 'sensor.nspanel_entryway_notification_label', 'sensor.nspanel_entryway_notification_text', 'number.nspanel_entryway_display_brightness', 'number.nspanel_entryway_display_brightness_dimdown', 'number.nspanel_entryway_display_brightness_sleep', 'number.nspanel_entryway_temperature_correction', 'number.nspanel_entryway_timeout_page', 'number.nspanel_entryway_timeout_dimming', 'number.nspanel_entryway_timeout_sleep', 'select.nspanel_entryway_baud_rate', 'select.nspanel_entryway_wake_up_page', 'select.nspanel_entryway_update_tft_display_model', 'select.nspanel_entryway_update_tft_display_baud_rate', 'light.nspanel_entryway_relay_2', 'sensor.nspanel_entryway_device_name'], 'device_name': 'nspanel_entryway', 'device_name_sensor': 'sensor.nspanel_entryway_device_name', 'left_button': 'binary_sensor.nspanel_entryway_left_button', 'right_button': 'binary_sensor.nspanel_entryway_right_button', 'relay01_entity': 'switch.nspanel_entryway_relay_1', 'relay02_entity': 'switch.nspanel_entryway_relay_2', 'blueprint_entity': 'binary_sensor.nspanel_entryway_blueprint', 'currentpage': 'sensor.nspanel_entryway_current_page', 'detailed_entity': 'sensor.nspanel_entryway_detailed_entity', 'nextion_inited': 'binary_sensor.nspanel_entryway_nextion_display', 'nspaneltemp': 'sensor.nspanel_entryway_temperature', 'notification_text': 'sensor.nspanel_entryway_notification_text', 'notification_unread': 'switch.nspanel_entryway_notification_unread', 'thermostat_embedded': 'climate.nspanel_entryway_thermostat', 'home_page_background_update': True}, 'trigger_data': {'id': 'home_custom_buttons', 'idx': '91', 'alias': None}}" result=ScriptRunResu...f06812235f0>})>.
```

**Comment by Der-Sven** [comment]
Can confirm this behaviour.

HA crashes and reboots.

**Comment by eddiebullman** [comment]
Had the same issue but I rolled HA back to 2024.3.3

**Comment by edwardtfn** [comment]
Could you please update your Home Assistant to v2024.4.2 and let me know if this issue persists?

**Comment by russilker** [comment]
Just updated, issue persists.

**Comment by bdraco** [comment]
`RuntimeError: no running event loop` means you have an integration doing non-threadsafe operations on asyncio objects which is corrupting the internal asyncio state.

When an integration modifies the event loop internals from another thread the behavior is undefined and will usually eventually lead to a crash.

If you put [asyncio in debug mode](https://docs.python.org/3/library/asyncio-dev.html#debug-mode), it will be able to detect some of these bad calls and maybe even block them.  You can use the profiler service to do that:
https://www.home-assistant.io/integrations/profiler/#service-profilerset_asyncio_debug

If you can't get far enough along you can also enable `debugpy:` in `configuration.yaml` instead (but its much slower)
https://www.home-assistant.io/integrations/debugpy/

**Comment by edwardtfn** [comment]
I will wait for a fix from Alarmo side before continue with this.

**Comment by edwardtfn** [comment]
I'm closing this as I understood this was an issue with Alarmo and was fixed with Alarmo v1.10.0.
